scripts/redcap-extract.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+import requests
+import io
+import csv
+from ftplib import FTP
+from datetime import datetime
+
+def extract_to_bytes(extract, fields=None):
+    """ Converts the REDCAP API response data to a io bytes handle
+    Paramters
+    _________
+    extract: list of dicts
+    fields: list of str
+        Field names to include in CSV.
+
+    Returns
+    _______
+    fb : io.BytesIO
+
+    """
+    return iostr_to_iobytes(write_csv(io.StringIO(''), extract, fields))
+
+def write_csv(f, extract, fields=None):
+    """ Writers extract to file handle
+    Parameters
+    __________
+    f : file handle (string mode)
+    extract: list of dicts
+    fields: list of str
+        Field names to include in CSV.
+
+    Returns
+    _______
+    f : file handle (string mode)
+    """
+    keys = fields if fields is not None else extract[0].keys()
+    dict_writer = csv.DictWriter(f, keys, extrasaction='ignore')
+    dict_writer.writeheader()
+    dict_writer.writerows(extract)
+    return f
+
+def iostr_to_iobytes(fs):
+    """ Converts io.StringIO object to io.BytesIO
+    Paramters
+    _________
+    fs : io.StringIO
+
+    Returns
+    _______
+    fb : io.BytesIO
+    """
+    fs.flush()
+    fs.seek(0)
+    fb = io.BytesIO(fs.read().encode())
+    return fb
+
+def bytes_to_ftp(f, project, name, ip, username='', password=''):
+    """ Saves file to the REDCAP extracts folder on RADAR-FTP.
+    Parameters
+    __________
+    f : file handle (bytes)
+    project : string
+        Name of the project folder to save under
+    name : string
+        Name to give the file
+    """
+    ftp = FTP(ip)
+    if (username and password):
+        ftp.login(username, password)
+    ftp.storbinary('STOR /RADAR-CNS/REDCAP/{}/{}'.format(project, name), f)
+    ftp.close()
+    return
+
+def save_extract_ftp(extract, ip, project, *args, **kwargs):
+    date = datetime.now()
+    name = 'REDCAP_{}_{}.csv'.format(project, date.strftime('%Y%m%d_%H%M'))
+    fields = kwargs.pop('fields', None)
+    f = extract_to_bytes(extract, fields)
+    bytes_to_ftp(f, project, name, ip, *args, **kwargs)
+    return
+
+def call_api(url, payload, **kwargs):
+    r = requests.post(url, data=payload, **kwargs)
+    r.raise_for_status()
+    return r.json()
+
+def get_redcap_metadata(url, token):
+    payload = {'token': token,
+               'format': 'json',
+               'content': 'metadata',
+               'type': 'flat'}
+    return call_api(url, payload)
+
+def get_redcap_extract(url, token):
+    payload = {'token': token,
+               'format': 'json',
+               'content': 'record',
+               'type': 'flat',
+               'exportSurveyFields': 'true'}
+    return call_api(url, payload)
+
+def field_filter(entry, exclude_notes=False):
+    blacklist_fields = ('landline', 'giffgaff_password', 'fitbit_password')
```

Comment:
are these not marked as identifiable data in RedCap? If not, we should probably do.

Reply:
I think for RADAR-CNS they are not rather dummy data and email addresses used to register the devices, but for other scenarios they may be using participant devices in which case they should be flagged as `identifiable fields`. I think safer in the future probably to do this -- would need to go in with another update to the RADAR-DDs. @speelen might oblige if still available.

Reply:
we added these as flagged identifiable since this now so I think this is ok.